src/afrolabs/components/kafka.clj

```diff
@@ -1930,20 +2203,28 @@ Supports a timeout operation. `timeout-duration` must be a java.time.Duration.")
                                                                 caught-up-ch)]))
 
         ktable-state (atom {})
+        merge-updates-opts (cond-> {}
+                             retention-ms (assoc :retention-ms retention-ms))
         consumer-client (reify
                           IConsumerClient
                           (consume-messages
                               [_ msgs]
                             (when (seq msgs)
                               (swap! ktable-state
-                                     #(merge-updates-with-ktable % msgs)))))
+                                     #(merge-updates-with-ktable % msgs merge-updates-opts)))))
+
+        retention-ms-seek-back-strategy (when retention-ms
+                                          (let [seek-to-timestamp (t/- (-time/get-current-time clock)
+                                                                       (t/duration retention-ms))]
+                                            (SeekToTimestampOffset seek-to-timestamp)))
```

Comment:
if we specify `:retention-ms` then we may also limit the amount of data loaded initially with a seek to an appropriate timestamp